src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -1440,11 +1440,15 @@ def is_ne_structure(self, other: 'ModelSystem') -> bool:
 
     def is_root_system(self) -> bool:
         """
-        A node is root if its parent is not a ModelSystem section.
-        Prefer isinstance for robustness; fall back to m_def identity.
+        True if this node has no parent or its parent is not a ModelSystem.
+        Prefer an ``isinstance`` check; fall back to comparing ``m_def`` to handle
+        proxy/wrapper parents that still expose a ModelSystem definition.
         """
 
-        return (self.m_parent is None) or (self.m_parent.m_def is not ModelSystem.m_def)
+        parent = self.m_parent
+        return (parent is None) or not (
+            isinstance(parent, ModelSystem) or parent.m_def is ModelSystem.m_def
```

Comment:
does the `is` work when comparing objects or `==` is preferred?

Reply:
what is know is `is` if you really compare objects in memory, in general objects should be compared with `==`, anyway this is just a minor thing. indeed `is` is less error prone as we do not have to go through the `__eq__` check, but metainfo will not allow defining this so they are basically the same. For point 2, sorry yes indeed you are right, it will return True for everything other than ModelSystem